src/unittests/floating_point.spec.ts

```diff
@@ -7574,135 +7574,169 @@ interface MatrixScalarToMatrixCase {
   expected: (number | IntervalBounds)[][];
 }
 
+const kMultiplicationMatrixScalarIntervalCases = {
+  f32: [
+    // From https://github.com/gpuweb/cts/issues/3044
+    {
+      matrix: [
+        [kValue.f32.negative.min, 0],
+        [0, 0],
+      ],
+      scalar: kValue.f32.negative.subnormal.min,
+      expected: [
+        [[0, reinterpretU32AsF32(0x407ffffe)], 0], // [[0, 3.9999995...], 0],
+        [0, 0],
+      ],
+    },
+  ] as MatrixScalarToMatrixCase[],
+  f16: [
+    // From https://github.com/gpuweb/cts/issues/3044
+    {
+      matrix: [
+        [kValue.f16.negative.min, 0],
+        [0, 0],
+      ],
+      scalar: kValue.f16.negative.subnormal.min,
+      expected: [
+        [[0, reinterpretU16AsF16(0x43fe)], 0], // [[0, 3.99609375], 0]
+        [0, 0],
+      ],
+    },
+  ] as MatrixScalarToMatrixCase[],
+} as const;
+
 g.test('multiplicationMatrixScalarInterval')
   .params(u =>
     u
       .combine('trait', ['f32', 'f16'] as const)
       .beginSubcases()
-      .combineWithParams<MatrixScalarToMatrixCase>([
+      .expandWithParams<MatrixScalarToMatrixCase>(p => {
         // Only testing that different shapes of matrices are handled correctly
         // here, to reduce test duplication.
         // multiplicationMatrixScalarInterval uses MultiplicationIntervalOp for calculating intervals,
-        // so the testing for multiplcationInterval covers the actual interval
+        // so the testing for multiplicationInterval covers the actual interval
```

Comment:
So it seems that the assumption in this comment doesn't stand (being an implementation bug), because the bug is considered caused by misusing `multiplicationInterval` without calling it but actually call its implementation directly.

Maybe we can add to this comment with some note like `Some subnormal cases are also tested to ensure that inputs are correctly flushed when calling multiplicationInterval.`?

I am not sure what to do to help avoid similar problem. I looks through the `floating_point.ts`, and the only other interval implementation that calls other implementation (`.impl`) is `ldexpIntervalImpl`, and it explicitly handles the flushing (hopefully it does it right)

Reply:
I have added to the comment language calling out that additional testing is occuring, and pointing to the original bug as the reason why.

I am not sure of a good way to avoid these issues, since the fundamental issue from my perspective is that the various run*Op functions do manipulation of the inputs and then need to invoke the *Op.impl functions, but we want things that are not the run*Op functions to be prevented from invoking the impls.

Currently they are all members of the same class, so there isn't something simple like changing access qualifiers that can be done. We could decompose the FP classes into different functional units so that the API, Impl, and Run portions are in different classes, so that we could use OO relationships to limit accidental misuse. I am not sure off hand if all of that refactoring and additional complexity is worth it.

I am not sure in TS if there is a better way to control the invokability of the impls. Like being able to inspect the calling function to make sure it is a run*Op (I know older JS allowed you to reflect the caller in a function, but I am pretty sure that is deprecated.)

Reply:
I think ldexp is fine, since it is explicitly invoking the roundAndFlush code that run*Op would.